packages/nodes-base/nodes/RenameKeys/RenameKeys.node.ts

```diff
@@ -78,23 +86,30 @@ export class RenameKeys implements INodeType {
 		const returnData: INodeExecutionData[] = [];
 
 		let item: INodeExecutionData;
-		let newItem: INodeExecutionData;
+		let keepOnlyRenamed: boolean;
 		let renameKeys: IRenameKey[];
 		let value: any; // tslint:disable-line:no-any
 		for (let itemIndex = 0; itemIndex < items.length; itemIndex++) {
+			keepOnlyRenamed = this.getNodeParameter('keepOnlyRenamed', itemIndex, false) as boolean;
 			renameKeys = this.getNodeParameter('keys.key', itemIndex, []) as IRenameKey[];
 			item = items[itemIndex];
-
-			// Copy the whole JSON data as data on any level can be renamed
-			newItem = {
-				json: JSON.parse(JSON.stringify(item.json)),
+			
+			const newItem: INodeExecutionData = {
+				json: {},
+			};
+			
+			if (keepOnlyRenamed !== true) {
+				if (item.binary !== undefined) {
+					// Reference binary data if any exists. We can reference it
+					// as this nodes does not change it
+					newItem.binary = item.binary;
+				}
+				
+				// Copy the whole JSON data as data on any level can be renamed
+				newItem = {
```

Comment:
You're re-assigning a constant here. L97 should use the `let` keyword, or the original L81 should be brought back and both L97 and L109 should just be bare assignments.

Reply:
Thanks, I messed up there. 
Fixed and tested it.